src/alert/alert.ts

```diff
@@ -85,7 +92,12 @@ export class NgbDismissibleAlert implements OnInit {
       this.closeEvent.emit($event);
       this.close();
     });
+    if (this.dismissOnTimeout) {
+      this._timeout = setTimeout(() => { this.close(); }, toInteger(this.dismissOnTimeout));
```

Comment:
Here is a fixed plunk: http://plnkr.co/edit/yLx44InWGCX2jbsx5C6l?p=preview

Surprisingly it seems like it works in a constructor!